sympy/stats/random_matrix_models.py

```diff
@@ -212,6 +214,45 @@ def joint_eigen_distribution(mat):
     """
     return mat.pspace.model.joint_eigen_distribution()
 
+@dispatch(Matrix)
```

Comment:
are you going to use this function here only? If so, no need to dispatch. Multiple dispatching is only needed if you dispatch over different files.

Reply:
Yes, I will apply, https://github.com/sympy/sympy/pull/17306#discussion_r309337663, so we will not required to use `dispatch`.